app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java

```diff
@@ -124,9 +128,44 @@ public void run() {
         //todo: where to start service? login activity?
         //WakefulIntentService.sendWakefulWork(this, GGService.actionGetTipsIntent(this));
 
+        // Check if the GPS is enabled,
+        // and ask the user if he wants to enable it
+        checkIfGpsEnabled();
     }
 
-    private void CalculateCurrentLocation() {
+    private void checkIfGpsEnabled() {
+        final LocationManager manager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+
+        if (!manager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
+            buildAlertMessageNoGps();
+        }
+    }
+
+    private void buildAlertMessageNoGps() {
```

Comment:
We would like to clean MainActivity as possible.
please extract this method to other class. maybe create new directory for dialogs. I guess that we will have more from them.